### PR TITLE
add: direct README

### PR DESCRIPTION
**Pull request by PhilYue**

### CODE REVIEW

File: direct/README.md [hunk 1]
```diff
@@ -0,0 +1,156 @@
+# Direct Example
+
+### Backend
+
+In the development and testing environment, it is often necessary to bypass the registry and test the designated service provider, which may require point-to-point direct connection. The point-to-point direct connection method will be based on the service interface and ignore the list of providers in the registry. Interface A is configured point-to-point and does not affect interface B to get the list from the registry.
+
+This example provides the `Consumer` point-to-point direct connection `Provider` based on Dubbo-Go to complete service calls to help better understand the connectivity of Dubbo-Go.
+
+### 目录
```
Comment (AlexStocks) [inline direct/README.md:9]: hey, guy, pls translate the chinese into english.

File: direct/README.md [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+        └── integration
+```
+- go-server: The Service Provider
+- go-client: The Service Consumer
+
+#### Provider
+Direct example code description:
+
+1. Configure the Dubbo protocol, registry, service information, See [server.yml](go-server/conf/server.yml)
+```yaml
```
Comment (AlexStocks) [inline direct/README.md:31]: pls add a blank line btw line 30 and line 31

File: direct/README.md [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+    protocol: "dubbo"
+    interface: "org.apache.dubbo.UserProvider"
+    cluster: "failover"
+    # this is necessary to enable direct-invoking.
+    url: "dubbo://127.0.0.1:20000"
+    methods:
+      - name: "GetUser"
+        retries: 3
+```
+2. Startup: Direct connection to the service to complete a service call
```
Comment (AlexStocks) [inline direct/README.md:117]: add a blank line above line 117 and lint 118

File: direct/go-client/conf/client.yml [hunk 4]
```diff
@@ -37,25 +28,25 @@ references:
...
-      tcp_w_buf_size: 65536
-      pkg_rq_size: 1024
-      pkg_wq_size: 512
-      tcp_read_timeout: "1s"
-      tcp_write_timeout: "5s"
-      wait_timeout: "1s"
-      max_msg_len: 1024000
-      session_name: "client"
\ No newline at end of file
+#protocol_conf:
```
Comment (beiwei30) [inline direct/go-client/conf/client.yml:31]: don't comment out, if it's unnessary, simply remove it.

**Review by beiwei30 — CHANGES_REQUESTED** [review]

**Review by AlexStocks — APPROVED** [review]

**Review by LaurenceLiZhixin — APPROVED** [review]

**Review by LaurenceLiZhixin — APPROVED** [review]

**Review by zouyx — APPROVED** [review]

**Pull request merged**